dolphin/alert_manager/alert_processor.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+
+from dolphin.drivers import manager as driver_manager
+from dolphin import db
+
+LOG = log.getLogger(__name__)
+
+
+class AlertProcessor(object):
+    """Alert model translation and export functions"""
+    def __init__(self):
+        self.driver_manager = driver_manager.DriverManager()
+
+    def process_alert_info(self, alert):
+        """Fills alert model using driver manager interface."""
+
+        # Get storage_id using source ip address
+        # If device wants to use source ip different from registration ip, it can be done as part of alert source config
+
+        # context to be made use of in future
+        context = {}
+
+        # First retrieve access_info from source ip and get storage info using storage id
+        filters = {'host': alert['transport_address']}
+        access_info = db.access_info_get_all(context, filters=filters)
+
+        # For given source ip, there should be unique access_info
+        if len(access_info) != 1:
```

Comment:
It probably not correct to check if len(access_info) != 1 here, take vmax as example, unisphere may manage several vmax devices at same time, they have the same host, while have different extra_attributes, so when you get the access_info by host you will get several items.  And different device may have different situation. So I suggest to let driver to decide how to get the specific access_info.

Reply:
Updated in comments

Reply:
if host is required to find the storage. I suggest we move it to alert_trap_config.  IMO, the ip in access info may not be same with alert server ip.

Reply:
Yes, when we handle alert config flow, this will be taken care